### PR TITLE
Parse `intrinsic_type_spec` in the declaration

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Fixes: #1605

### CODE REVIEW

File: tests/attributes1.f90 [hunk 1]
```diff
@@ -17,5 +17,6 @@ program attributes1
 end type details
 intrinsic sin, cos
 doubleprecision, intent (in) :: x(..)
+type(real(kind=4)), allocatable :: x(:)
```
Comment (certik) [inline tests/attributes1.f90:20]: Test all the types that you support in the parser, just to be sure (integer, logical, etc.)

**Comment by certik** [comment]
Awesome, thanks for fixing this!

File: tests/attributes1.f90 (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply tests/attributes1.f90:20]: Done!

**Review by certik — APPROVED** [review]
Awesome, thanks! I checked manually that the grammar is still LALR(1) when the symbol conflicts are removed. So all good.

**Pull request merged**